internal/app/synchronizer/synchronizer_service.go

```diff
@@ -110,6 +111,13 @@ func (s *synchronizerService) Synchronize(stream ipb.Synchronizer_SynchronizeSer
 				registration.allM1cSent.Done()
 				return
 			}
+			if _, ok := matchIDs[req.GetProposal().GetMatchId()]; ok {
```

Comment:
This has a couple problems:

It's not detecting conflicts from multiple FetchMatches calls.  This synchronize method only sees one.

We most definitely needs to return an error here, instead of just logging.